loopy/check.py

```diff
@@ -1207,6 +1207,135 @@ def check_that_shapes_and_strides_are_arguments(kernel):
 # }}}
 
 
+# {{{ validate_kernel_call_sites
+
+def _are_sub_array_refs_equivalent(sar1, sar2, caller):
+    """
+    Returns *True* iff *sar1* and *sar2* are equivalent
+    :class:`loopy.SubArrayRef`s.
+
+    Two sub-array-refs are said to be equivalent iff they point to the same
+    array sub-regions. This equivalence check is less stricter than
+    :meth:`~loopy.SubArrayRef.is_equal`.
+
+    :arg caller: An instance of :class:`loopy.LoopKernel` in which they are
+         referenced.
+    """
+    if len(sar1.swept_inames) != len(sar2.swept_inames):
+        return False
+
+    if sar1.subscript.aggregate.name != sar2.subscript.aggregate.name:
+        return False
+
+    if len(sar1.subscript.index_tuple) != len(sar2.subscript.index_tuple):
+        return False
+
+    from loopy.symbolic import SubstitutionMapper
+    from pymbolic.mapper.substitutor import make_subst_func
+    from pymbolic.mapper.distributor import distribute
+    subst_func = make_subst_func({iname1.name:  iname2
+                                  for iname1, iname2 in zip(sar1.swept_inames,
+                                                            sar2.swept_inames)
+                                  })
+
+    for sweep1, sweep2 in zip(sar1.swept_inames, sar2.swept_inames):
+        sweep1_bounds = caller.get_iname_bounds(sweep1.name)
+        sweep2_bounds = caller.get_iname_bounds(sweep2.name)
+        if (sweep1_bounds.lower_bound_pw_aff != sweep2_bounds.lower_bound_pw_aff):
+            return False
+
+        if (sweep1_bounds.upper_bound_pw_aff != sweep2_bounds.upper_bound_pw_aff):
+            return False
```

Comment:
Compare the access ranges instead. Computing bounds is extremely expensive.

Reply:
Yep, makes sense: 521a2c0ee61ba9c4f92ef80029f6e284d08cb490